e2e/workspace/src/create-nx-workspace.test.ts

```diff
@@ -242,4 +242,40 @@ describe('create-nx-workspace', () => {
     checkFilesDoNotExist('package-lock.json');
     process.env.SELECTED_PM = packageManager;
   });
+
+  describe('Use detected package manager', () => {
+    const wsName = uniq('pm');
+
+    function setupProject(envPm: 'npm' | 'yarn' | 'pnpm') {
+      process.env.SELECTED_PM = envPm;
+      runCreateWorkspace(wsName, {
+        preset: 'empty',
+        packageManager: envPm,
+        useDetectedPm: true,
+      });
+    }
+
+    afterEach(() => {
+      process.env.SELECTED_PM = packageManager;
+    });
+
+    it('should use npm when invoked with npx', () => {
+      setupProject('npm');
+      checkFilesExist(`package-lock.json`);
+      checkFilesDoNotExist(`yarn.lock`, `pnpm-lock.yam`);
+    }, 90000);
+
+    it('should use pnpm when invoked with pnpx', () => {
+      setupProject('pnpm');
+      checkFilesExist(`pnpm-lock.yaml`);
+      checkFilesDoNotExist(`yarn.lock`, `package-lock.json`);
+    }, 90000);
+
+    // skipping due to packageManagerCommand for createWorkspace not using yarn create nx-workspace
```

Comment:
Any idea why it doesn't?

Reply:
![image](https://user-images.githubusercontent.com/25516557/129278199-2ddb1148-b55b-42f5-b16b-99a8c5f0d4bb.png)
There's this comment in `getPackageManagerCommand`.